GitFyle.Core.Api.Tests.Unit/Services/Foundations/Contributions/ContributionServiceTests.Validations.Add.cs

```diff
@@ -0,0 +1,345 @@
+        {
+            // given
+            DateTimeOffset randomDateTime = GetRandomDateTimeOffset();
+            DateTimeOffset now = randomDateTime;
+            Contribution randomContribution = CreateRandomContribution(now);
+            Contribution invalidContribution = randomContribution;
+            invalidContribution.ExternalCreatedAt = now;
+            invalidContribution.ExternalUpdatedAt = GetRandomDateTimeOffset();
+
+            var invalidContributionException = new InvalidContributionException(
+                message: "Contribution is invalid, fix the errors and try again.");
+
+            invalidContributionException.AddData(
+                key: nameof(Contribution.ExternalUpdatedAt),
+                values: $"Date is not the same as {nameof(Contribution.ExternalCreatedAt)}");
+
+            var expectedContributionValidationException =
+                new ContributionValidationException(
+                    message: "Contribution validation error occurred, fix errors and try again.",
+                    innerException: invalidContributionException);
+
+            this.dateTimeBrokerMock.Setup(broker =>
+                broker.GetCurrentDateTimeOffsetAsync())
+                    .ReturnsAsync(now);
+
+            // when
+            ValueTask<Contribution> addContributionTask =
+                this.contributionService.AddContributionAsync(invalidContribution);
+
+            ContributionValidationException actualContributionValidationException =
+                await Assert.ThrowsAsync<ContributionValidationException>(
+                    addContributionTask.AsTask);
+
+            // then
+            actualContributionValidationException.Should().BeEquivalentTo(
+                expectedContributionValidationException);
+
+            this.dateTimeBrokerMock.Verify(broker =>
+                broker.GetCurrentDateTimeOffsetAsync(),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogErrorAsync(It.Is(
+                    SameExceptionAs(expectedContributionValidationException))),
+                        Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.InsertContributionAsync(It.IsAny<Contribution>()),
+                    Times.Never);
+
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Theory]
+        [InlineData(-11)]
+        public async Task ShouldThrowValidationExceptionOnAddIfAuditPropertiesIsNotRecentAndLogItAsync(
+           int invalidYears)
+        {
+            // given
+            DateTimeOffset randomDateTime =
+             GetRandomDateTimeOffset();
+
+            DateTimeOffset now = randomDateTime;
+            DateTimeOffset startDate = now.AddYears(-10);
```

Comment:
You don't have to do a start date / end date.  Use the invalidSeconds parameter.  It will then test each scenario based on that InlineData value that is being injected.
Have a look at the example done on the SourceService